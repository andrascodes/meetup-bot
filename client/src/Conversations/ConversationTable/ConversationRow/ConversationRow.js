import React from 'react'

import { formatDate, calculateDuration } from '../../../lib/dateManipulation'

export const ConversationRow = props => (
  <tr id={props.id} onClick={props.handleRowClick}>
    <td>{props.participant}</td>
    <td>{formatDate(props.start)}</td>
    <td>{calculateDuration(props.start, props.end)}</td>
    <td>{props.label}</td>
    <td>{props.errors}</td>
  </tr>
)

ConversationRow.PropTypes = {
  id: React.PropTypes.number,
  handleRowClick: React.PropTypes.func.isRequired,
  participant: React.PropTypes.string.isRequired,
  start: React.PropTypes.string.isRequired,
  end: React.PropTypes.string.isRequired,
  errors: React.PropTypes.string,
  label: React.PropTypes.string
}