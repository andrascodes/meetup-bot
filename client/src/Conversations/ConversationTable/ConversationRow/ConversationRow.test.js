import React from 'react'
import renderer from 'react-test-renderer'

import { ConversationRow } from './ConversationRow'

describe('ConversationRow', () => {

  test('renders correctly', () => {
    const tree = renderer.create(
      <ConversationRow 
        id={1}
        handleRowClick={() => {}}
        participant={'123'}
        start={'2017-04-13 12:00'}
        end={'2017-04-13 12:02'}
        errors={'OAuthException'}
        label={'label1 label2'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})