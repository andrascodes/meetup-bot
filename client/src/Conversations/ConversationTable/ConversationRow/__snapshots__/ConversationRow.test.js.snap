exports[`ConversationRow renders correctly 1`] = `
<tr
  id={1}
  onClick={[Function]}>
  <td>
    123
  </td>
  <td>
    April 13th 2017 12:00
  </td>
  <td>
    02:00
  </td>
  <td>
    label1 label2
  </td>
  <td>
    OAuthException
  </td>
</tr>
`;
